### PR TITLE
Fix propagation of inherited variables in selection script

### PR DESCRIPTION
**Pull request by luispinedamx**
This fix also solves partially the propagation of variables in clean scripts.
Related to issue ow2-proactive/studio#410
The additions to functional test `Job_selection_script_variables.xml` are to validate all 6 possible scenarios of inherited variable propagation.

### CODE REVIEW

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue

1. MAJOR [InternalTask.java#L1196](https://github.com/ow2-proactive/scheduling/blob/3dfc3ae9cb43703b95874357d266009bc7f81383/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L1196): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS134)

**Comment by luispinedamx** [comment]
retest this please

**Comment by sophiesongge** [comment]
retest this please

**Comment by fviale** [comment]
Retest this please

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java [hunk 1]
```diff
@@ -745,6 +745,7 @@ public synchronized void prepareTasks() {
         tasks.clear();
         // re-init taskId
         int id = 0;
+        this.jobInfo.setVariables(this.getVariablesAsReplacementMap());
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java]: Very good find ! It all comes to the fact that the jobInfo is not initialized when the job is submitted (created using the empty no arg constructor of JobInfoImpl). The best here would be to properly initialize it, which is partially done in jobSubmitted.submitAction().

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue

1. MAJOR [InternalTask.java#L1196](https://github.com/ow2-proactive/scheduling/blob/32ec061219d1de88128f33ef6d4a3f6c9edd777c/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L1196): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS134)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue

1. MAJOR [InternalTask.java#L1196](https://github.com/ow2-proactive/scheduling/blob/04b05549a5672961f30784669d993ed8e57d917b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L1196): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS134)

**Pull request merged**